### PR TITLE
Make algolia comment search work for subforums, topic discussion, events, groups

### PR DESCRIPTION
**Pull request by Will-Howard**
Previously it only worked for posts.

Another way of doing this would be to put the url in the search hit, although that would introduce the possibility of the results getting out of sync if we change the url structure, which is why I didn't do that. I wouldn't mind changing to that way if anyone knows of any problems with the way I've done it (e.g. it costs more, adding more fields degrades the search quality)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203092851132284) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/search/algoliaDocuments.d.ts [hunk 1]
```diff
@@ -19,7 +19,13 @@ interface AlgoliaComment {
   postId?: string,
   postTitle?: string,
   postSlug?: string,
+  postSequenceId?: string,
```
Comment (s-cheng) [inline packages/lesswrong/server/search/algoliaDocuments.d.ts:22]: I don't think this is being used anywhere else?

File: packages/lesswrong/components/search/CommentsSearchHit.tsx [hunk 2]
```diff
@@ -39,12 +42,24 @@ const CommentsSearchHit = ({hit, clickAction, classes, showIcon=false}: {
...
+      groupId: comment.postGroupId,
+    })}#${comment._id}`;
+  }
+
   return <div className={classes.root}>
     {showIcon && <LWTooltip title="Comment">
       <ChatBubbleOutlineIcon className={classes.icon}/>
     </LWTooltip>}
-    <Link to={url} onClick={(event: React.MouseEvent) => isLeftClick(event) && clickAction && clickAction()}>
+    <Link to={url} onClick={(event: React.MouseEvent) => url && isLeftClick(event) && clickAction && clickAction()}>
```
Comment (s-cheng) [inline packages/lesswrong/components/search/CommentsSearchHit.tsx:62]: Why this change? I think sometimes the `clickAction` is not routing to a new page, but instead something like selecting this item, which I don't think we'd want to prevent?

**Review by s-cheng — APPROVED** [review]
ASTLAC

Very exciting! 🙌

File: packages/lesswrong/components/search/CommentsSearchHit.tsx (hunk 2 above)
Comment (Will-Howard) [reply packages/lesswrong/components/search/CommentsSearchHit.tsx:62]: Ah hmm I wasn't aware of that. This was to prevent it routing to the wrong page if the url couldn't be constructed, but I guess that should be the responsibility of `clickAction`, I'll change this back to how it was

File: packages/lesswrong/server/search/algoliaDocuments.d.ts (hunk 1 above)
Comment (Will-Howard) [reply packages/lesswrong/server/search/algoliaDocuments.d.ts:22]: it's used in `postGetPageUrl`
```
// Get URL of a post page.
export const postGetPageUrl = function(post: PostsMinimumForGetPageUrl, isAbsolute=false, sequenceId:string|null=null): string {
  const prefix = isAbsolute ? getSiteUrl().slice(0,-1) : '';

  // LESSWRONG – included event and group post urls
  if (sequenceId) {
    return `${prefix}/s/${sequenceId}/p/${post._id}`;
  } else if (post.isEvent) {
    return `${prefix}/events/${post._id}/${post.slug}`;
  } else if (post.groupId) {
    return `${prefix}/g/${post.groupId}/p/${post._id}/`;
  }
  return `${prefix}/posts/${post._id}/${post.slug}`;
};
```

Comment (s-cheng) [reply packages/lesswrong/server/search/algoliaDocuments.d.ts:22]: But `CommentsSearchHit` doesn't pass in `postSequenceId` when it calls `postGetPageUrl()`? Also we don't populate `postSequenceId` in Algolia so it would always be null anyway?
